### ISSUE TITLE
Fuzzer repeats itself

### ISSUE THREAD

**Issue by holiman** [ethereum/evmlab#113]
With the latest changes, it seems to sometimes happen that it goes into some sort of loop, generating the same cases over and over. 
Example printout: 
```
> python3 utilities/fuzzerweb.py  -c statetests.ini 2>&1 | grep Processed | grep geth
[...]
2018-11-05 14:40:34,202 - INFO - Processed 1934 steps for geth on test martin-Mon_14_39_55-24670-120, pTime:224.87 ms 
2018-11-05 14:40:34,452 - INFO - Processed 502 steps for geth on test martin-Mon_14_39_55-24670-140, pTime:12.51 ms 
2018-11-05 14:40:34,469 - INFO - Processed 2 steps for geth on test martin-Mon_14_39_55-24670-141, pTime:0.19 ms 
2018-11-05 14:40:34,470 - INFO - Processed 33 steps for geth on test martin-Mon_14_39_55-24670-143, pTime:0.61 ms 
2018-11-05 14:40:34,756 - INFO - Processed 1934 steps for geth on test martin-Mon_14_39_55-24670-121, pTime:225.50 ms 
2018-11-05 14:40:34,947 - INFO - Processed 38 steps for geth on test martin-Mon_14_39_55-24670-144, pTime:1.15 ms 
2018-11-05 14:40:34,950 - INFO - Processed 38 steps for geth on test martin-Mon_14_39_55-24670-145, pTime:0.99 ms 
2018-11-05 14:40:35,302 - INFO - Processed 33 steps for geth on test martin-Mon_14_39_55-24670-147, pTime:1.16 ms 
2018-11-05 14:40:35,507 - INFO - Processed 3 steps for geth on test martin-Mon_14_39_55-24670-148, pTime:0.36 ms 
2018-11-05 14:40:35,820 - INFO - Processed 33 steps for geth on test martin-Mon_14_39_55-24670-149, pTime:1.09 ms 
2018-11-05 14:40:35,936 - INFO - Processed 3 steps for geth on test martin-Mon_14_39_55-24670-150, pTime:0.31 ms 
2018-11-05 14:40:36,082 - INFO - Processed 33 steps for geth on test martin-Mon_14_39_55-24670-151, pTime:0.99 ms 
2018-11-05 14:40:36,256 - INFO - Processed 38 steps for geth on test martin-Mon_14_39_55-24670-152, pTime:1.07 ms 
2018-11-05 14:40:36,703 - INFO - Processed 1934 steps for geth on test martin-Mon_14_39_55-24670-126, pTime:239.95 ms 
2018-11-05 14:40:36,948 - INFO - Processed 502 steps for geth on test martin-Mon_14_39_55-24670-153, pTime:19.20 ms 
2018-11-05 14:40:36,987 - INFO - Processed 502 steps for geth on test martin-Mon_14_39_55-24670-154, pTime:19.56 ms 
2018-11-05 14:40:37,009 - INFO - Processed 2 steps for geth on test martin-Mon_14_39_55-24670-156, pTime:0.24 ms 
2018-11-05 14:40:37,047 - INFO - Processed 502 steps for geth on test martin-Mon_14_39_55-24670-155, pTime:18.98 ms 
2018-11-05 14:40:37,210 - INFO - Processed 3 steps for geth on test martin-Mon_14_39_55-24670-157, pTime:0.31 ms 
2018-11-05 14:40:37,327 - INFO - Processed 38 steps for geth on test martin-Mon_14_39_55-24670-158, pTime:0.80 ms 
2018-11-05 14:40:37,328 - INFO - Processed 2 steps for geth on test martin-Mon_14_39_55-24670-159, pTime:0.17 ms 
2018-11-05 14:40:37,416 - INFO - Processed 2 steps for geth on test martin-Mon_14_39_55-24670-160, pTime:0.24 ms 
2018-11-05 14:40:37,538 - INFO - Processed 33 steps for geth on test martin-Mon_14_39_55-24670-161, pTime:1.00 ms 
2018-11-05 14:40:37,634 - INFO - Processed 3 steps for geth on test martin-Mon_14_39_55-24670-162, pTime:0.29 ms 
2018-11-05 14:40:37,722 - INFO - Processed 33 steps for geth on test martin-Mon_14_39_55-24670-163, pTime:0.77 ms 
2018-11-05 14:40:38,002 - INFO - Processed 33 steps for geth on test martin-Mon_14_39_55-24670-165, pTime:1.31 ms 
2018-11-05 14:40:38,154 - INFO - Processed 0 steps for geth on test martin-Mon_14_39_55-24670-166, pTime:0.23 ms 
2018-11-05 14:40:38,253 - INFO - Processed 2 steps for geth on test martin-Mon_14_39_55-24670-167, pTime:0.27 ms 
2018-11-05 14:40:38,372 - INFO - Processed 502 steps for geth on test martin-Mon_14_39_55-24670-164, pTime:18.58 ms 
2018-11-05 14:40:38,484 - INFO - Processed 33 steps for geth on test martin-Mon_14_39_55-24670-168, pTime:1.18 ms 
2018-11-05 14:40:38,641 - INFO - Processed 2 steps for geth on test martin-Mon_14_39_55-24670-169, pTime:0.27 ms 
2018-11-05 14:40:38,643 - INFO - Processed 38 steps for geth on test martin-Mon_14_39_55-24670-171, pTime:1.18 ms 
2018-11-05 14:40:39,038 - INFO - Processed 3 steps for geth on test martin-Mon_14_39_55-24670-173, pTime:0.27 ms 
2018-11-05 14:40:39,196 - INFO - Processed 2 steps for geth on test martin-Mon_14_39_55-24670-174, pTime:0.31 ms 
2018-11-05 14:40:39,354 - INFO - Processed 502 steps for geth on test martin-Mon_14_39_55-24670-172, pTime:20.41 ms 
2018-11-05 14:40:39,509 - INFO - Processed 2 steps for geth on test martin-Mon_14_39_55-24670-175, pTime:0.28 ms 
2018-11-05 14:40:39,741 - INFO - Processed 33 steps for geth on test martin-Mon_14_39_55-24670-176, pTime:1.13 ms 
2018-11-05 14:40:40,656 - INFO - Processed 502 steps for geth on test martin-Mon_14_39_55-24670-178, pTime:21.54 ms 
2018-11-05 14:40:40,836 - INFO - Processed 38 steps for geth on test martin-Mon_14_39_55-24670-180, pTime:1.09 ms 
2018-11-05 14:40:41,200 - INFO - Processed 502 steps for geth on test martin-Mon_14_39_55-24670-179, pTime:22.85 ms 
2018-11-05 14:40:41,500 - INFO - Processed 33 steps for geth on test martin-Mon_14_39_55-24670-181, pTime:1.18 ms 
2018-11-05 14:40:41,766 - INFO - Processed 38 steps for geth on test martin-Mon_14_39_55-24670-183, pTime:1.13 ms 
2018-11-05 14:40:42,211 - INFO - Processed 1934 steps for geth on test martin-Mon_14_39_55-24670-142, pTime:251.95 ms 
2018-11-05 14:40:42,522 - INFO - Processed 33 steps for geth on test martin-Mon_14_39_55-24670-184, pTime:1.20 ms 
2018-11-05 14:40:42,525 - INFO - Processed 33 steps for geth on test martin-Mon_14_39_55-24670-185, pTime:1.20 ms 
2018-11-05 14:40:42,758 - INFO - Processed 33 steps for geth on test martin-Mon_14_39_55-24670-186, pTime:1.27 ms 
2018-11-05 14:40:43,309 - INFO - Processed 1934 steps for geth on test martin-Mon_14_39_55-24670-146, pTime:265.82 ms 
2018-11-05 14:40:43,608 - INFO - Processed 502 steps for geth on test martin-Mon_14_39_55-24670-188, pTime:19.60 ms 
2018-11-05 14:40:43,627 - INFO - Processed 38 steps for geth on test martin-Mon_14_39_55-24670-189, pTime:0.69 ms 
2018-11-05 14:40:43,629 - INFO - Processed 2 steps for geth on test martin-Mon_14_39_55-24670-190, pTime:0.26 ms 
2018-11-05 14:40:44,306 - INFO - Processed 38 steps for geth on test martin-Mon_14_39_55-24670-192, pTime:1.29 ms 
2018-11-05 14:40:44,497 - INFO - Processed 502 steps for geth on test martin-Mon_14_39_55-24670-191, pTime:17.63 ms 
2018-11-05 14:40:44,646 - INFO - Processed 2 steps for geth on test martin-Mon_14_39_55-24670-193, pTime:0.29 ms 
2018-11-05 14:40:44,805 - INFO - Processed 2 steps for geth on test martin-Mon_14_39_55-24670-194, pTime:0.29 ms 
2018-11-05 14:40:44,953 - INFO - Processed 38 steps for geth on test martin-Mon_14_39_55-24670-195, pTime:1.27 ms 
2018-11-05 14:40:45,126 - INFO - Processed 33 steps for geth on test martin-Mon_14_39_55-24670-196, pTime:1.23 ms 
2018-11-05 14:40:45,282 - INFO - Processed 2 steps for geth on test martin-Mon_14_39_55-24670-197, pTime:0.26 ms 
2018-11-05 14:40:45,438 - INFO - Processed 2 steps for geth on test martin-Mon_14_39_55-24670-198, pTime:0.30 ms 
2018-11-05 14:40:45,592 - INFO - Processed 33 steps for geth on test martin-Mon_14_39_55-24670-199, pTime:1.08 ms 
2018-11-05 14:40:45,724 - INFO - Processed 2 steps for geth on test martin-Mon_14_39_55-24670-200, pTime:0.33 ms 
2018-11-05 14:40:45,882 - INFO - Processed 33 steps for geth on test martin-Mon_14_39_55-24670-201, pTime:1.21 ms 
2018-11-05 14:40:46,100 - INFO - Processed 38 steps for geth on test martin-Mon_14_39_55-24670-202, pTime:1.22 ms 
2018-11-05 14:40:46,609 - INFO - Processed 2 steps for geth on test martin-Mon_14_39_55-24670-205, pTime:0.25 ms 
2018-11-05 14:40:46,903 - INFO - Processed 502 steps for geth on test martin-Mon_14_39_55-24670-203, pTime:21.94 ms 
2018-11-05 14:40:47,094 - INFO - Processed 38 steps for geth on test martin-Mon_14_39_55-24670-206, pTime:1.16 ms 
2018-11-05 14:40:47,550 - INFO - Processed 1934 steps for geth on test martin-Mon_14_39_55-24670-170, pTime:249.02 ms 
2018-11-05 14:40:47,838 - INFO - Processed 38 steps for geth on test martin-Mon_14_39_55-24670-207, pTime:1.14 ms 
2018-11-05 14:40:48,033 - INFO - Processed 2 steps for geth on test martin-Mon_14_39_55-24670-208, pTime:0.19 ms 
2018-11-05 14:40:48,334 - INFO - Processed 502 steps for geth on test martin-Mon_14_39_55-24670-209, pTime:21.11 ms 
2018-11-05 14:40:48,997 - INFO - Processed 1934 steps for geth on test martin-Mon_14_39_55-24670-177, pTime:255.12 ms 
2018-11-05 14:40:49,312 - INFO - Processed 502 steps for geth on test martin-Mon_14_39_55-24670-210, pTime:21.83 ms 
2018-11-05 14:40:49,358 - INFO - Processed 502 steps for geth on test martin-Mon_14_39_55-24670-212, pTime:21.89 ms 
2018-11-05 14:40:49,610 - INFO - Processed 38 steps for geth on test martin-Mon_14_39_55-24670-214, pTime:1.22 ms 
2018-11-05 14:40:50,176 - INFO - Processed 1934 steps for geth on test martin-Mon_14_39_55-24670-182, pTime:255.95 ms 
2018-11-05 14:40:50,518 - INFO - Processed 38 steps for geth on test martin-Mon_14_39_55-24670-215, pTime:1.24 ms 

```
You can see certain patterns, `1934` steps, and `502` and `38` recurring. I believe it somehow regenerates the same pieces of code over and over again.

**Comment by holiman** [comment]
Also, it seems to almost never go into any real `depth`. Try a local fuzzerweb and compare with the one in prod, which goes a lot more into calls than the current head.

**Comment by holiman** [comment]
Maybe the repetition is in fact the christmas-fuzzing, and is to be expected? (i.e : not actual repetition, but with minor changes)

**Comment by tintinweb** [comment]
that should be fixed with #116 plus we've added some statistics (top 10 traceDepth distribution) with #120
